### PR TITLE
Update Hazelcast example

### PR DESCRIPTION
**Pull request by pires**
- Added support for different DNS domain
- Took care of nits detailed in #7444 
- Updated documentation
- Added `Dockerfile` for future takeover on the image maintenance

/cc @brendandburns

### CODE REVIEW

**Comment by pires** [comment]
Of course they trust me!

File: examples/hazelcast/README.md [hunk 1]
```diff
@@ -78,18 +78,23 @@ spec:
         - resources:
             limits:
               cpu: 1
-          image: pires/hazelcast-k8s:0.2
+          image: quay.io/pires/hazelcast-kubernetess:0.3.1
```
Comment (brendandburns) [inline examples/hazelcast/README.md]: I think you mean kubernetes:0.3.1 not kubernetess (kubernetesssssssssss!)

**Comment by brendandburns** [comment]
Other than the typo, LGTM.  No need to build the PR, as nothing is changing that is built.

**Comment by pires** [comment]
@brendandburns done.

**Comment by brendandburns** [comment]
LGTM.

**Pull request merged**